### ISSUE TITLE
fix: `shorebird` commands do not check out the correct version of flutter.

### ISSUE THREAD

**Issue by eseidel** [shorebirdtech/shorebird#399]
I just installed `shorebird` on windows.   I'm now not sure what to do to fix, my doctor is sad:

```
PS C:\Users\micro\flutter> shorebird doctor

Shorebird v0.0.9
Shorebird Engine • revision 687371bd24346cb3cad647d278afb7817da51c62
✗ AndroidManifest.xml files contain INTERNET permission (12ms)
  [✗] No Android project found
✓ Shorebird is up-to-date (0.7s)
✗ Flutter install is correct (7.4s)
  [!] Directory: 'C:\Users\micro\.shorebird\bin\cache\flutter' is not on the correct revision

2 issues detected.
```

**Comment by eseidel** [comment]
```
PS C:\Users\micro\flutter> shorebird doctor

Shorebird v0.0.9
Shorebird Engine • revision 687371bd24346cb3cad647d278afb7817da51c62
✗ AndroidManifest.xml files contain INTERNET permission (12ms)
  [✗] No Android project found
✓ Shorebird is up-to-date (0.7s)
✗ Flutter install is correct (7.4s)
  [!] Directory: 'C:\Users\micro\.shorebird\bin\cache\flutter' is not on the correct revision

2 issues detected.
PS C:\Users\micro\flutter> cd C:\Users\micro\.shorebird\bin\cache\flutter
PS C:\Users\micro\.shorebird\bin\cache\flutter> .\bin\flutter --version
Flutter 3.7.10-0.0.pre.7 • channel unknown • unknown source
Framework • revision 5cf5567f89 (24 hours ago) • 2023-04-26 13:20:54 -0700
Engine • revision 687371bd24
Tools • Dart 2.19.6 • DevTools 2.20.1
```

**Comment by eseidel** [comment]
Maybe this is just https://github.com/shorebirdtech/shorebird/issues/364?

**Comment by eseidel** [comment]
This comes from: https://github.com/shorebirdtech/shorebird/blob/main/bin/shorebird.ps1#L19 not respecting bin/internal/flutter.version.  

https://github.com/shorebirdtech/shorebird/blob/main/third_party/flutter/bin/internal/shared.sh#L16 doesn't either.  oops.

**Comment by bryanoltman** [comment]
Update: the correct revision of flutter is checked out, the problem was with the version string not being trimmed 🤦 

Fixed by #401
